File: slidepager/src/main/java/com/liangfeizc/screenslidepager/SlidePagerAdapter.java
package com.liangfeizc.screenslidepager;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.facebook.common.internal.Lists;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by liangfeizc on 3/26/15.
 */
public class SlidePagerAdapter extends FragmentStatePagerAdapter {
    private List<String> picList = Lists.newArrayList();

    public SlidePagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return SlidePageFragment.newInstance(picList.get(i));
    }

    @Override
    public int getCount() {
        return picList.size();
    }

    public void addAll(List<String> picList) {
        this.picList = picList;
    }
}

File: slidepager/src/main/java/com/liangfeizc/slidepager/SlidePagerAdapter.kt
package com.liangfeizc.slidepager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

/**
 * Created by liangfei on 3/26/15.
 */
class SlidePagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    private var picList = mutableListOf<String>()

    override fun getItem(i: Int): Fragment {
        return SlidePageFragment.newInstance(picList[i])
    }

    override fun getCount() = picList.size

    fun addAll(picList: List<String>) {
        this.picList.apply {
            clear()
            addAll(picList)
        }
    }
}
